core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -260,6 +271,19 @@ public SparkSqlDialect(SqlDialect.Context context) {
       default:
         return super.getTargetFunc(call);
       }
+    case OTHER_FUNCTION:
+      switch (call.type.getSqlTypeName()) {
```

Comment:
Please create a separate method for **OTHER_FUNCTION** handling.

Reply:
done. created.